vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -318,25 +375,49 @@ func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 		for _, mount := range containerJSONInfo.Mounts {
 			// check if the mount location belongs to vsphere plugin
 			if matchNameforVMDK(mount.Source) {
-				r.Incr(mount.Name)
-				log.Debugf("  name=%v (driver=%s source=%s) (%v)",
+				// populate temp map
+				rc := auxRefMap[mount.Name]
```

Comment:
this looks way too complicated. I thought the idea was 
* set Dirty
* start map discovery, with timeout and retries
* if dirty: 
   * no refcounts on mount
   * no unmount, ,just return success
* when map is properly build
   * unmount hanging stuff at the end
    * clear dirty flag

What is done now does not seem to be any different as far as locks and races, but uses 2 maps for some reason. Can you elaborate why the first map is even needed when Dirty ?